pkgs/http/lib/src/client.dart

```diff
@@ -34,6 +35,13 @@ import 'streamed_response.dart';
 /// another instance of [Client] and add functionality on top of that. This
 /// allows all classes implementing [Client] to be mutually composable.
 abstract interface class Client {
+  /// Whether this client supports the [RequestController] API.
+  ///
+  /// If this is `true`, [send] will use the supplied [RequestController] and
+  /// will allow cancelling the request and specifying a timeout. Otherwise,
+  /// a specified [RequestController] will be ignored.
+  bool get supportsController;
```

Comment:
This is a breaking change for `implements Client` uses. For example https://github.com/google/googleapis.dart/blob/ef216bc000480b20d8a08776343bd2f8eacd423d/googleapis_auth/lib/src/auth_client.dart#L10

Reply:
Oops - as was already the case with `BaseClient` that was supposed to return `false` by default to allow clients to opt-in.

Reply:
Adding the API at all is breaking. Any class which `implement Client` without `extends BaseClient` will be missing the override.

Reply:
Oh yes, of course. I'll do some testing against something like the linked client (`AuthClient`) for that then - I didn't initially find any clients that implemented `Client` so I presumed they would use `BaseClient` like `ReplyClient` does.

That might need further review then, but off the top of my head a backwards-compatible solution could be to introduce some stub that could be mixed in or included and tested for.

For example: https://dartpad.dev/bc9a693d707f95a0356291817f38db1e